src/Documentation/Grain-Versioning/Grain-versioning.md

```diff
@@ -0,0 +1,35 @@
+# Grain versioning
+
+> [!WARNING]
+> This page describes how to use grain interface versioning. The versioning of
+> Grain state it out of the scope.
+
+## Overview
+On a given cluster, silos can support different versions of a grain type.
+![Cluster with different versions of a grain](version.png)
+In this example the client and Silo{1,2,3} were compiled with grain interface `A` version 1. Silo 4 was compiled with `A` version 2.
+
+## Limitations:
+-	No versioning on stateless worker
+-	Streaming agents are not versioned
+
+## Enable versioning
+By default, grains are not versioned. You can version grain by using the VersionAttribute on the grain interface:
+
+``` cs
+[Version(X)]
+public interface IVersionUpgradeTestGrain : IGrainWithIntegerKey {}
+```
+
+Where `X` is the version of the grain interface.
```

Comment:
Theoratically, you could start at version 999 and decrease the number for each new version, but it doesn't seems very natural to me. But since we could do it anyway (rollback previous version for instance) I think the suggestion from @jdom is valid